clients/src/main/java/org/apache/kafka/clients/producer/internals/BuiltInPartitioner.java

```diff
@@ -279,6 +279,13 @@ public int partition() {
         }
     }
 
+    /*
+     * Default hashing function to choose a partition from the serialized key bytes
+     */
+    public static int partitionForKey(final byte[] serializedKey, final int numPartitions) {
```

Comment:
The intention of introducing this static function is to make sure everyone (including streams)'s default partitioner behavior is the same; and in the future if we ever change the algorithm we would make sure we change everyone as well.